kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/MessageIdUtils.java

```diff
@@ -107,6 +107,14 @@ public static final long offsetAfterBatchIndex(long offset) {
         return offset;
     }
 
+    public static long getHighWaterMark(ManagedLedger managedLedger) {
```

Comment:
Hi @aloyszhang Seems these 2 methods are the same, we provided a wrapper on getCurrentOffset, but the inside code not changed, Would you please help provide more info on how could this fix the issue?

Reply:
`getCurrentOffset`  gives the max offset for the current time, and `getHighWaterMark` and `getLogEndOffset` will give  `getCurrentOffset` + 1.